tests/FACE/GetConnectionParameters/Subscriber/Subscriber.cpp

```diff
@@ -0,0 +1,201 @@
+#endif
+
+#include "ace/OS_NS_unistd.h"
+#include <iostream>
+
+bool callbackHappened = false;
+int callback_count = 0;
+long expected = 0;
+FACE::CONNECTION_ID_TYPE connId;
+bool callback_unregistered = false;
+
+void callback(FACE::TRANSACTION_ID_TYPE,
+              Messenger::Message& msg,
+              FACE::MESSAGE_TYPE_GUID message_type_id,
+              FACE::MESSAGE_SIZE_TYPE message_size,
+              const FACE::WAITSET_TYPE,
+              FACE::RETURN_CODE_TYPE& return_code)
+{
+  ++callback_count;
+  expected = msg.count + 1;
+  ACE_DEBUG((LM_INFO, "In callback() (the %d time): %C\t%d\t"
+             "message_type_id: %Ld\tmessage_size: %d\n",
+             callback_count, msg.text.in(), msg.count,
+             message_type_id, message_size));
+  callbackHappened = true;
+  if (callback_count % 10 == 0) {
+    ACE_DEBUG((LM_INFO, "Subscriber: about to Unregister_Callback()\n"));
+    FACE::TS::Unregister_Callback(connId, return_code);
+    if (return_code != FACE::RC_NO_ERROR) return;
+    callback_unregistered = true;
+  }
+  return_code = FACE::RC_NO_ERROR;
+}
+
+int ACE_TMAIN(int, ACE_TCHAR*[])
+{
+  FACE::RETURN_CODE_TYPE status = FACE::RC_NO_ERROR;
+  FACE::TS::Initialize("face_config.ini", status);
+  FACE::MESSAGE_SIZE_TYPE size;
+
+  if (!status) {
+    FACE::CONNECTION_DIRECTION_TYPE dir;
+    FACE::TS::Create_Connection("sub", FACE::PUB_SUB, connId, dir, size, FACE::INF_TIME_VALUE, status);
+  }
+
+  if (!status) {
+    FACE::CONNECTION_NAME_TYPE name = {};
+    FACE::TRANSPORT_CONNECTION_STATUS_TYPE connectionStatus;
+    FACE::TS::Get_Connection_Parameters(name, connId, connectionStatus, status);
+    if (status != FACE::NOT_AVAILABLE) {
+      std::cout << "ERROR: Get_Connection_Parameters after creating connection returned " << status
+                << " not FACE::NOT_AVAILABLE (" << FACE::NOT_AVAILABLE << ")" << std::endl;
+      return EXIT_FAILURE;
+    }
+    status == FACE::RC_NO_ERROR;
+  }
+
+  ACE_DEBUG((LM_INFO, "Subscriber: about to Register_Callback()\n"));
+  FACE::TS::Register_Callback(connId, 0, callback, 0, status);
+  if (status != FACE::RC_NO_ERROR) return static_cast<int>(status);
+
+  bool post_reg_callback_check_ok = true;
+  if (!status) {
+    FACE::CONNECTION_NAME_TYPE name = {};
+    FACE::TRANSPORT_CONNECTION_STATUS_TYPE connectionStatus;
+    FACE::TS::Get_Connection_Parameters(name, connId, connectionStatus, status);
+    if (status != FACE::RC_NO_ERROR) {
+      std::cout << "ERROR: Get_Connection_Parameters after registering callback returned " << status << std::endl;
+      return static_cast<int>(status);
+    }
+    if (connectionStatus.CONNECTION_DIRECTION != FACE::DESTINATION ||
+        connectionStatus.LAST_MSG_VALIDITY != FACE::INVALID ||
+        connectionStatus.MESSAGE != 1 ||
+        connectionStatus.MAX_MESSAGE != ACE_INT32_MAX ||
+        connectionStatus.MAX_MESSAGE_SIZE != 300 ||
+        connectionStatus.WAITING_PROCESSES_OR_MESSAGES != 0 ||
```

Comment:
We need one test where WAITING_PROCESSES_OR_MESSAGES  is expected to be non-zero also.